### PR TITLE
bolt info implementation

### PR DESCRIPTION
**Pull request by rajzshkr**
- bolt info package-name
- bolt info package-name --json

### CODE REVIEW

File: src/utils/yarn.js [hunk 1]
```diff
@@ -122,3 +122,11 @@ export async function bin(cwd: string) {
     tty: true
   });
 }
+
+export async function info(cwd: string, dependencies: Array<string> = []) {
```
Comment (thejameskyle) [inline src/utils/yarn.js]: `dependencies`?

The arguments to `yarn info` are `yarn info [pkg] <key>`:

```js
$ yarn info lodash
{ name: 'lodash',
  description: 'Lodash modular utilities.',
  'dist-tags': { latest: '4.17.4' },
  versions:
   [ '0.1.0',
     '0.2.0',
     '0.2.1',
     '0.2.2',
     '0.3.0',
     '0.3.1',
     '0.3.2',
     '0.4.0',
     '0.4.1'
     ... }
$ yarn info lodash versions
[ '0.1.0',
  '0.2.0',
  '0.2.1',
  '0.2.2',
  '0.3.0',
  '0.3.1',
  '0.3.2',
  '0.4.0',
  '0.4.1',
  ... ]
```

We should make the signature `info(cwd: string, pkg: string, key?: string)`

File: src/commands/info.js [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
 
 export function toInfoOptions(
   args: options.Args,
   flags: options.Flags
 ): InfoOptions {
-  return {};
+  const depsArgs = Array.prototype.concat([], args);
+  Object.keys(flags).forEach(key => {
+    if (flags[key] === true) {
+      depsArgs.push(`--${key}`);
```
Comment (thejameskyle) [inline src/commands/info.js]: We shouldn't convert flags like this. We should explicitly support individual flags

File: src/utils/yarn.js (hunk 1 above)
Comment (rajzshkr) [reply src/utils/yarn.js]: We've to send an array of arguments to the spawn process (https://github.com/boltpkg/bolt/blob/master/src/utils/processes.js#L67). Renamed to `dependencies` to `spawnArgs`.

File: src/commands/info.js (hunk 2 above)
Comment (rajzshkr) [reply src/commands/info.js]: > We should explicitly support individual flags

I didn't understand this.  I've seen the same kind of implementation in the project https://github.com/boltpkg/bolt/blob/master/src/utils/yarn.js#L16.

**Comment by rajzshkr** [comment]
@thejameskyle Updated the PR. I' recently started work on boltpkg, I take reference from the project to implement new feature. If something is not correct, please give your comments. Always happy to make changes.

File: src/commands/info.js (hunk 2 above)
Comment (thejameskyle) [reply src/commands/info.js]: That's a very explicit conversion. I'm saying we shouldn't accept arbitrary flags and pass them through

File: src/commands/info.js [hunk 3]
```diff
@@ -1,16 +1,45 @@
...
-export type InfoOptions = {};
+export type InfoOptions = {|
+  cwd?: string,
+  args: options.Args
+|};
+
+function flagsToArgs(flagArgs: options.Flags): Array<string> {
+  const flagArgsKey = Object.keys(flagArgs);
+  // Flags has default -- key
+  if (flagArgsKey.length < 1) {
```
Comment (ajaymathur) [inline src/commands/info.js]: It should be `<= 1`  🙂

File: src/commands/info.js [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+    return [];
+  }
+  const flag = [];
+  flagArgsKey.forEach(key => {
+    if (flagArgs[key] === true) {
+      flag.push(`--${key}`);
```
Comment (ajaymathur) [inline src/commands/info.js]: flags received here are converted to camelCase by meow, therefore, they may not pass down to yarn same as what bolt receives, if json is the only flag supported (AFAIK) we can capture it explicitly.

Comment (ajaymathur) [inline src/commands/info.js]: flags received here are converted to camelCase by meow, therefore, they may not pass down to yarn same as what bolt receives, if json is the only flag supported (AFAIK) we can capture it explicitly.

File: src/commands/info.js (hunk 3 above)
Comment (rajzshkr) [reply src/commands/info.js]: @ajaymathur I removed this piece of code.

File: src/commands/info.js (hunk 4 above)
Comment (rajzshkr) [reply src/commands/info.js]: Done !!

**Pull request merged**